src/main.rs

```diff
@@ -46,126 +50,219 @@ impl LogLevel {
+        log::error!("Greenboot health-check failed!");
+        return Err(anyhow!("health-check failed!"));
     }
-    if failure {
-        for path in ["/etc/greenboot/red.d/*.sh"] {
-            for entry in glob(path)?.flatten() {
-                log::info!("running red check {}", entry.to_string_lossy());
-                let output = Command::new("bash").arg("-C").arg(entry).output()?;
-                if !output.status.success() {
-                    // combine and print stderr/stdout
-                    log::warn!("red script failed...");
-                }
+    Ok(())
+}
+
+fn run_red() -> Result<(), Error> {
+    for path in GREENBOOT_INSTALL_PATHS {
+        let red_path = format!("{}{}", path, "/red.d/*.*");
+        for entry in glob(&red_path)?.flatten() {
+            log::info!("running red check {}", entry.to_string_lossy());
+            let output = Command::new("bash").arg("-C").arg(entry.clone()).output()?;
+            if !output.status.success() {
+                // combine and print stderr/stdout
+                log::warn!("red script: {} failed!", entry.to_string_lossy());
             }
         }
-        log::warn!("SYSTEM is UNHEALTHY. Rolling back and rebooting...");
-        Command::new("rpm-ostree").arg("rollback").status()?;
-        reboot()?;
-        return Ok(());
     }
-    for path in ["/etc/greenboot/green.d/*.sh"] {
-        for entry in glob(path)?.flatten() {
+    Ok(())
+}
+
+fn run_green() -> Result<(), Error> {
+    for path in GREENBOOT_INSTALL_PATHS {
+        let green_path = format!("{}{}", path, "/green.d/*.*");
+        for entry in glob(&green_path)?.flatten() {
             log::info!("running green check {}", entry.to_string_lossy());
-            let output = Command::new("bash").arg("-C").arg(entry).output()?;
+            let output = Command::new("bash").arg("-C").arg(entry.clone()).output()?;
             if !output.status.success() {
                 // combine and print stderr/stdout
-                log::warn!("green script failed...");
+                log::warn!("green script {} failed!", entry.to_string_lossy());
             }
         }
     }
     Ok(())
 }
 
-fn reboot() -> Result<(), Error> {
-    Command::new("systemctl").arg("reboot").spawn()?;
-    Ok(())
-}
+fn health_check() -> Result<()> {
+    let config = get_config().unwrap();
+    log::info!("{config:?}");
 
-fn stamp() -> Result<(), Error> {
-    fs::create_dir_all("/etc/greenboot/")?;
-    File::create("/etc/greenboot/upgrade.stamp")?;
-    Ok(())
+    let run_status = run_diagnostics();
+    match run_status {
+        Ok(()) => {
+            log::info!("greenboot health-check passed.");
+            run_green().expect("Green scripts failed!");
+            handle_motd("green");
+            handle_boot_success(true).expect("unable to set grub variable");
+            Ok(())
+        }
+        Err(e) => {
+            handle_motd("red");
+            run_red().expect("Red scripts failed!");
```

Comment:
Its better to use ? than .expect , since it will cause program to panic on error